### ISSUE TITLE
“Robomongo” can’t be opened because it is from an unidentified developer.

### ISSUE THREAD

**Issue by hemanthkandukuri** [paralect/robomongo#963]
When i try to install on Mac OS X Yosemite, am getting this error. Even though i downloaded a genuine copy of Robomongo from robomongo.org but still not able to install it.

Can you please look into the issue

Regards, 
Hemanth

**Comment by robbie-c** [comment]
As a temporary workaround, you might be able to open it by right-clicking it then clicking "open". This will give you the same warning about it being from an unidentified developer, but will give you the option to proceed anyway.

**Comment by mcpDESIGNS** [comment]
Or go into System Preference -> Security & Privacy
Allows apps downloaded from:
 * Anywhere

**Comment by juliashibalko** [comment]
https://github.com/paralect/robomongo/issues/792

**Comment by jeroenvanagt** [comment]
It seems that version 0.9.0 RC9 still is not signed on OSX. When do you expect a signed version to be available?

**Comment by simsekgokhan** [comment]
Hi @jeroenvanagt , thanks a lot for your interest in Robomongo and sorry for the delay. This is a major candidate for next releases. Currently, we are focused on some major features like Replica Sets, Import/Export. We hope we will include this feature soon.

**Comment by stanchan** [comment]
Any ideas when this will be worked on? This prevents Robomongo from being installed in a Corporate environment, due to security policies preventing unsigned applications from being installed.

**Comment by simsekgokhan** [comment]
Hi @stanchan , we have put this into our "critical" list. As I understand (and can you confirm) you cannot use Robomongo even with workarounds because you cannot change security settings? Or hopefully you can at least use it with some hacky way for now until we provide signed installers?

**Comment by simsekgokhan** [comment]
And regarding update about investigation, although procedure of signing seems trivial, it will require some file/folder base structural re-arrangements (due to Apple signing requirements) in the project which in return will probably require code changes. (at lease in cmake files).

Next Action:
- Plan to include this in a new release

**Comment by cheat117** [comment]
Hey @simsekgokhan I can confirm that if your enterprise does not allow you access to manipulate the security settings, you will be unable to use the native installer not will you be able to use the linux installer (tried it, same issue). Short of Virtualboxing a copy of linux, there is no hacky method for me to use the software.

Thanks!

**Comment by stanchan** [comment]
I'm confirming as well... without the ability to access the security settings, there is no way to launch the robomongo application due to Mac's App Sandbox.

**Comment by simsekgokhan** [comment]
Thanks a lot for confirmation @cheat117 @stanchan. 
And what happens when you perform previously suggested solution? It opens it without installing as I understand.
> As a temporary workaround, you might be able to open it by right-clicking it then clicking "open"

**Comment by mennogroen** [comment]
The workaround (opening through right-click) is not an option in an enterprise setting, where the ability to bypass the signature check is disabled.

**Comment by simsekgokhan** [comment]
@mennogroen Thanks for the clarification.

**Comment by clockpenalty** [comment]
Confirmed: Robomongo cannot be used in an enterprise setting because it requires signing and the right click-open method requires root access which is not available.
This appears to be a serious problem especially for an enterprise tool.

**Comment by simsekgokhan** [comment]
Hi @clockpenalty , thanks for reporting the problem with details.
This issue is still in our priority list. Hope to provide fix soon.

**Comment by kzvaigzne** [comment]
@simsekgokhan It's been quite a while since this was opened, then it's been a while since it was put in your "critical" list and now it's been a while with 0 new developments regarding this as far as I know - **The issue is still there**!

--------
In terms of helping others with this, I've found this online and it fixes the issue with a workaround:

> OS X improves download validation by providing file quarantine in applications that download files from the Internet. This means that downloads are checked for safety (known malware) when you try to open them.
> You can check an app (or any file) in quarantine using terminal. It's literally an extended file system attribute:
> 
> mac:~ user$ xattr /Applications/Some.app
> com.apple.quarantine
> mac:~ user$
> 
> If you're absolutely sure it's safe to run the app, you can drop the attribute like follows (you'll be asked to provide your password):
> mac:~ user$ sudo xattr -r -d com.apple.quarantine /Applications/Some.app
> 
> Check once again:
> mac:~ user$ xattr /Applications/Some.app
> mac:~ user$
> 
> Now you should be able to run the app.

**Comment by Fallenstedt** [comment]
Lack of certificates still prevents Robomongo from being installed in a Corporate environment, due to security policies preventing unsigned applications from being installed. This issue has been here since Oct 28, 2015. How is it going?

**Comment by NealHumphrey** [comment]
Bump again, experiencing the same issue.

**Comment by hieutrandn9889** [comment]
It still occurs on Mac OS

**Comment by ashokramcse** [comment]
It still exists in macOS 10.13.5.

**Comment by nameisaravind** [comment]
“Robo 3T” can’t be opened because it is from an unidentified developer. - This still occured till date - 07/05/2018 in macOS High Sierra 10.13.4

**Comment by freelancersunil** [comment]
“Robo 3T” can’t be opened because it is from an unidentified developer. Your security preferences allow installation of only apps from the App Store and identified developers.
Chrome downloaded this file today at 8:13 PM from robomongo.org.

Please fix this - High Sierra  10.13.4

**Comment by cheat117** [comment]
you can open the applications directory, right click the app, and choose open. that's my work around until there is a signed package

**Comment by alkesh** [comment]
This is still an issue on Mojave (10.14.1)

**Comment by jacob87o2** [comment]
Can't believe it's still not resolved after more than 3 years. This should be top priority. 😞

**Comment by sirisaacnuketon** [comment]
Wow still not looking into this? How can something marked as critical still not be dealt with over 3 years later? Guess I'll just find an alternative as I literally have no way of opening it on a corporate machine.

**Comment by crkepler** [comment]
I just downloaded Robomongo (Robo 3T) and due to this issue, I had no choice but to drag it to the Mac's trash can. Looking for an alternative. Since the developers don't seem to care (3+ years), neither do I.

**Comment by nabaraj** [comment]
Any other alternative GUI tools for mac os as robomongo/ robo3t will not work as I can see it is taking 3+yrs to resolve this issue?

**Comment by david-robinson-practiceweb** [comment]
For anyone who can't install robo3t then you could try using the community edition of compass
https://www.mongodb.com/download-center/compass?jmp=docs

Does anyone know if studio3t is actually signed properly? If not then that's a potential blocker for a lot of prospects.

**Comment by kaos** [comment]
I agree. This can impossibly be a big issue to resolve. And given that the years starts to add up since this was reported, I can't take this project seriously. 

Going to look for alternatives. :/

**Comment by joquijada** [comment]
> @simsekgokhan It's been quite a while since this was opened, then it's been a while since it was put in your "critical" list and now it's been a while with 0 new developments regarding this as far as I know - **The issue is still there**!
> 
> In terms of helping others with this, I've found this online and it fixes the issue with a workaround:
> 
> > OS X improves download validation by providing file quarantine in applications that download files from the Internet. This means that downloads are checked for safety (known malware) when you try to open them.
> > You can check an app (or any file) in quarantine using terminal. It's literally an extended file system attribute:
> > mac:~ user$ xattr /Applications/Some.app
> > com.apple.quarantine
> > mac:~ user$
> > If you're absolutely sure it's safe to run the app, you can drop the attribute like follows (you'll be asked to provide your password):
> > mac:~ user$ sudo xattr -r -d com.apple.quarantine /Applications/Some.app
> > Check once again:
> > mac:~ user$ xattr /Applications/Some.app
> > mac:~ user$
> > Now you should be able to run the app.

This worked like a charm. I tried various other suggestions but non worked: denied app would appear in General of tab of Security&Privacy and would allow me to open from there for one hour? Nope. Select the `Anywhere` option in Secutiry&Privay general tab? Nope (`Anyhwere` no where to be found). Thanks a lot for sharing this gem, I've added it to my MacOs notes.

**Comment by eyalroth** [comment]
> Any ideas when this will be worked on? This prevents Robomongo from being installed in a Corporate environment, due to security policies preventing unsigned applications from being installed.

Indeed, corporate environments often involve a policy enforcement / MDM product preventing installations of "unverified applications".

If you have `sudo` permissions, you should be able to temporarily disable the gatekeeper and install any app:
1. Disable the gatekeeper via `sudo spctl --master-disable`.
2. Install application.
3. Re-enable the gatekeeper via `sudo spctl --master-enable`.

**Comment by oleksii-molozhanov** [comment]
Here we are 1 year more and the devs still can't sign the tool ...
Unfortunately have to try something else. Will check out:
`$ brew cask install mongodb-compass-community`
